src/app/account.service.ts

```diff
@@ -71,6 +78,37 @@ export class AccountService {
     }
   }
 
+  getDerivedUser(publicKey: string, blockHeight: number): DerivedUserInfo{
+    const privateUser = this.getPrivateUsers()[publicKey];
+
+    if (!this.entropyService.temporaryEntropy.mnemonic) {
```

Comment:
If for some reason this method was called multiple times in a row (usually it shouldn't be) wouldn't the same entropy be reused? Might be a good idea to always call `setNewTemporaryEntropy`?

Reply:
Makes sense to me.